lib/experimental/class-gutenberg-rest-global-styles-revisions-controller.php

```diff
@@ -0,0 +1,363 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Global_Styles_Revisions_Controller class, inspired by WP_REST_Revisions_Controller.
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 6.3.0
+ */
+
+/**
+ * Core class used to access global styles revisions via the REST API.
+ *
+ * @since 6.3.0
+ *
+ * @see WP_REST_Controller
+ */
+class Gutenberg_REST_Global_Styles_Revisions_Controller extends WP_REST_Controller {
```

Comment:
I did start with extending `WP_REST_Revisions_Controller`, but global style revisions need special treatment, e.g., 

- JSON instead of rendered HTML
- We only need one route right now (to fetch some revisions for the UI)
- I wanted to add some custom properties/schema

The methods I had to overwrite contained just as much code as doing it from scratch. 

Happy to go back to extending `WP_REST_Revisions_Controller` if folks think it's better longer term.